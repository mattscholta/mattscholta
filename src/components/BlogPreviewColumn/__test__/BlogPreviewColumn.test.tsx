import * as React from 'react';
import { MemoryRouter } from 'react-router';
import { render, RenderResult } from '@testing-library/react';

import {
  BlogPreviewColumn,
  BlogPreviewColumnProps
} from '../BlogPreviewColumn';

describe('BlogPreviewColumn', () => {
  let props: BlogPreviewColumnProps;
  let wrapper: RenderResult;

  const createWrapper = () => (
    <MemoryRouter>
      <BlogPreviewColumn {...props} />
    </MemoryRouter>
  );

  beforeEach(() => {
    props = {
      className: '',

      post: {
        __typename: 'BlogPost',
        date: '',
        excerpt: '',
        URL: '',
        content: '',
        featured_image: '',
        modified: '',
        post_thumbnail: {
          URL: '',
          __typename: 'BlogPostThumb',
          height: 100,
          width: 100
        },
        slug: '',
        sticky: false,
        title: ''
      },

      small: false
    };
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
