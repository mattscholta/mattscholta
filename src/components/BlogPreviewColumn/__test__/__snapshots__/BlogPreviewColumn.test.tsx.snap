// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPreviewColumn matches snapshot 1`] = `
NodeList [
  <a
    class="component"
    data-ga-label=""
    href="/posts"
  >
    <img
      alt=""
      class="image"
      height="1250"
      loading="eager"
      title=""
      width="1880"
    />
    <div
      class="u-flex-grow"
    >
      <div
        class="u-flex heading"
      >
        <h2 />
      </div>
    </div>
    <span
      class="link"
    >
      Read More Â»
    </span>
  </a>,
]
`;
